dali/interface.py

```diff
@@ -9,68 +7,27 @@ class CommunicationError(Exception):
     pass
 
 
-class DaliServer(object):
-    """Communicate with daliserver
-    (https://github.com/onitake/daliserver)
+class ParameterError(Exception):
+    pass
 
-    NB this requires daliserver commit
-    90e34a0cd2945dc7a15681f11647e708f858521e or later.
+class DriverInterface(object):
+    """
+    Driver interface class
+    Contains the common definition of functions for variousdrivers
     """
 
-    def __init__(self, host="localhost", port=55825,
-                 multiple_frames_per_connection=False):
-        self._target = (host, port)
-        self._s = None
-        self._multiple_frames_per_connection = multiple_frames_per_connection
-
-    def __enter__(self):
-        if self._multiple_frames_per_connection:
-            self._s = socket.create_connection(self._target)
-        return self
-
-    def __exit__(self, *vpass):
-        print(vpass)
-        if self._multiple_frames_per_connection:
-            self._s.close()
-            self._s = None
-
-    def send(self, command):
-        if self._s:
-            s = self._s
-        else:
-            s = socket.create_connection(self._target)
-
-        assert isinstance(command, Command)
-        message = struct.pack("BB", 2, 0) + command.frame.pack
-
-        logging.info(u"command: {}{}".format(
-            command, " (twice)" if command.is_config else ""))
-
-        # Set a default result which may be used if the first send fails
-        result = "\x02\xff\x00\x00"
-
-        try:
-            s.send(message)
-            result = s.recv(4)
-            if command.is_config:
-                s.send(message)
-                result = s.recv(4)
-        except:
-            raise
-        finally:
-            if not self._s:
-                s.close()
-
-        response = self.unpack_response(command, result)
-
-        if response:
-            logging.info(u"  -> {0}".format(response))
-
-        return response
+    def send(self):
+        """
+        Send command, and returns its response
+        :return:
+        """
+        raise RuntimeError("Not implemented method")
```

Comment:
Also you might consider abc module for defining abstract classes, e,g, https://github.com/wichert/lingua/blob/master/src/lingua/extractors/__init__.py#L136